src/containers/Application/ApplicationWrapper.tsx

```diff
@@ -2,50 +2,66 @@ import React from 'react'
 import { Switch, Route } from 'react-router-dom'
 import { Header } from 'semantic-ui-react'
 
-import { NoMatch } from '../../components'
+import { Loading, NoMatch } from '../../components'
+import { useUserState } from '../../contexts/UserState'
+import useLoadApplication from '../../utils/hooks/useLoadApplicationNEW'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { FullStructure, User } from '../../utils/types'
 
 const ApplicationWrapper: React.FC = () => {
-  return (
+  const { pathname, query } = useRouter()
+  const { serialNumber } = query
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  const { error, isLoading, structure, template } = useLoadApplication({
+    serialNumber,
+    currentUser: currentUser as User,
+    networkFetch: true,
+  })
+
+  return isLoading ? (
+    <Loading />
+  ) : structure ? (
     <Switch>
-      <Route exact path="/applicationNEW/new">
-        <ApplicationCreateNew />
-      </Route>
       <Route exact path="/applicationNEW/:serialNumber">
-        <ApplicationStartNew />
+        <ApplicationStartNew serialNumber={serialNumber} structure={structure} />
       </Route>
       <Route exact path="/applicationNEW/:serialNumber/:sectionCode/Page:page">
-        <ApplicationPageNew />
+        <ApplicationPageNew serialNumber={serialNumber} structure={structure} />
```

Comment:
They shouldn't need a serial number, and the path should be just be `<Route path={`${path}/:sectionCode/Page:page`}>`

Reply:
I agree it does look tidier, but simply can't make it work like that. Don't know why, the problem is: `path` comes as `application/12345` when I just want to go to the start page. which is comparing now `path={path}` so all good! But when I try `application/12345/S1/Page1` that's the path, so it also renders start page and not ApplicationPath with compares `path={`${path}/:sectionCode/Page:page`}`. Do you know how to solve this?

Reply:
@nmadruga 
this seems to work: https://github.com/openmsupply/application-manager-web-app/compare/%23342-build-fullStructure-in-application-wrapper...342-use-path

(it doesn't quite use useRouter hook, probably should add 'path' from useRouteMatch to userRouter hook in the future) I am actually not sure why we've added useRouter hook

Reply:
Thanks! Added to the PR on: f8f874b